app/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/javalin/DomainsRegistrationTest.groovy

```diff
@@ -97,14 +103,166 @@ class DomainsRegistrationTest extends Specification {
+        TestApplicationContext.injectRegisteredImplementations()
+
+        when:
+        def actualDomainResponse = DomainsRegistration.authenticateRequest(new DomainRequest())
+
+        then:
+        actualDomainResponse == expectedDomainResponse
+    }
+
+    def "authenticateRequest unauthorized request unhappy path works"() {
+        given:
+        def expectedStatusCode = 401
+
+        def mockValidator = Mock(AuthRequestValidator)
+        mockValidator.isValidAuthenticatedRequest(_ as DomainRequest) >> false
+        TestApplicationContext.register(AuthRequestValidator, mockValidator)
+
+        TestApplicationContext.register(Formatter, Jackson.getInstance())
+
+        TestApplicationContext.injectRegisteredImplementations()
+
+        when:
+        def res = DomainsRegistration.authenticateRequest(new DomainRequest())
+        def actualStatusCode = res.getStatusCode()
+
+        then:
+        noExceptionThrown()
+        actualStatusCode == expectedStatusCode
+    }
+
+    def "authenticateRequest SecretRetrievalException unhappy path works"() {
+        given:
+        def expectedStatusCode = 500
+
+        def mockValidator = Mock(AuthRequestValidator)
+        mockValidator.isValidAuthenticatedRequest(_ as DomainRequest) >> { throw new SecretRetrievalException("internal error", new IllegalArgumentException()) }
+        TestApplicationContext.register(AuthRequestValidator, mockValidator)
+
+        TestApplicationContext.register(Formatter, Jackson.getInstance())
+
+        TestApplicationContext.injectRegisteredImplementations()
+
+        when:
+        def res = DomainsRegistration.authenticateRequest(new DomainRequest())
+        def actualStatusCode = res.getStatusCode()
+
+        then:
+        actualStatusCode == expectedStatusCode
+    }
+
+    def "authenticateRequest IllegalArgumentException unhappy path works"() {
+        given:
+        def expectedStatusCode = 500
+
+        def mockValidator = Mock(AuthRequestValidator)
+        mockValidator.isValidAuthenticatedRequest(_ as DomainRequest) >> { throw new IllegalArgumentException("internal error", new IllegalArgumentException()) }
+        TestApplicationContext.register(AuthRequestValidator, mockValidator)
+
+        TestApplicationContext.register(Formatter, Jackson.getInstance())
+
+        TestApplicationContext.injectRegisteredImplementations()
+
+        when:
+        def res = DomainsRegistration.authenticateRequest(new DomainRequest())
+        def actualStatusCode = res.getStatusCode()
+
+        then:
+        actualStatusCode == expectedStatusCode
+    }
+
+    def "processRequest happy path works"() {
+        given:
+        def isProtected = true
+        def request = new DomainRequest()
+        def handler =  { DomainRequest req ->
+            return new DomainResponse(200)
+        }
+        def expectedStatusCode = (handler as Function<DomainRequest, DomainResponse>).apply(request).statusCode
```

Comment:
That's one way to do it.  You can also do this...
```suggestion
        def expectedStatusCode = 200
        def handler =  { DomainRequest req ->
            return new DomainResponse(expectedStatusCode)
        }
```

Feel free to take this or leave this.